core/eolearn/core/utils/parsing.py

```diff
@@ -13,15 +13,32 @@
 """
 from __future__ import annotations
 
+import sys
 from itertools import repeat
-from typing import TYPE_CHECKING, Iterable, List, Optional, Sequence, Tuple, Union, cast
+from typing import TYPE_CHECKING, Dict, Iterable, List, Optional, Sequence, Tuple, Union, cast
 
 from ..constants import FeatureType
+from .types import EllipsisType
+
+if sys.version_info < (3, 8):
+    from typing_extensions import Literal
+else:
+    from typing import Literal  # pylint: disable=ungrouped-imports
```

Comment:
I'd prefer to do as little re-importing as possible, as well as avoid users importing Literal from `utils.types`, because this will be a normal import next year (once 3.7 is not supported anymore). But I do agree it's ugly.